### PR TITLE
MM-60478: Do not invalidate session cache on user added to channel

### PR DESCRIPTION
**Pull request by agnivade**
PR: https://github.com/mattermost/mattermost/pull/27204 fixed the issue
where `invalidateWebConnSessionCacheForUser` was not broadcasted across
the cluster. However, we became too conservative and started to invalidate
the session cache as well.

But the session cache only contains the session details for the user
and the team membership. There is no channel membership related info.

Therefore, we avoid that and improve performance as well.

https://mattermost.atlassian.net/browse/MM-60478

```release-note
NONE
```

### CODE REVIEW

File: server/public/model/cluster_message.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ const (
 	ClusterEventInvalidateCacheForChannel                   ClusterEvent = "inv_channel"
 	ClusterEventInvalidateCacheForChannelGuestCount         ClusterEvent = "inv_channel_guest_count"
 	ClusterEventInvalidateCacheForUser                      ClusterEvent = "inv_user"
-	ClusterEventInvalidateCacheForUserTeams                 ClusterEvent = "inv_user_teams"
+	ClusterEventInvalidateWebConnCacheForUser               ClusterEvent = "inv_user_teams"
```
Comment (agnivade) [inline server/public/model/cluster_message.go:19]: I renamed the constant name but kept the actual value to be the same to preserve the Grafana charts.

**Review by devinbinnie — APPROVED** [review]
I'm okay with this as long as we're confident that the message will always arrive to the user, given my other change that forces it per user now it *should* be fine.

We should test this with a custom HA instance with 3 nodes - 2 nodes was not reproducing consistently.

**Comment by agnivade** [comment]
> I'm okay with this as long as we're confident that the message will always arrive to the user, given my other change that forces it per user now it should be fine.

Yep, we aren't changing the message. Just invalidating 1 cache instead of 2.

> We should test this with a custom HA instance with 3 nodes - 2 nodes was not reproducing consistently.

Noted. Will update the JIRA test steps.

**Comment by agnivade** [comment]
Hey @DHaussermann - the test steps are in JIRA. This needs a 3 server HA setup. Please create an HA spinwick and let me know. The cloud platform team will scale it up to 3 nodes.

**Pull request merged**